src/lib.rs

```diff
@@ -102,6 +102,7 @@
 //! ```
 
 #![deny(unsafe_code, missing_docs)]
+#![allow(async_fn_in_trait)]
```

Comment:
I don't think we actually want this. However, fixing the warning is going to take writing a bunch of boilerplate, I'm afraid

Reply:
I agree and would leave it as-is.